File: projects/wpf/NProcGen/tests/PonderingProgrammer.Map2d.Tests/CoordTest.cs
using PonderingProgrammer.Map2d;
using Xunit;

namespace PonderingProgrammer.Map2dTests
{
    public class CoordTest
    {
        [Fact]
        public void TestEquality()
        {
            var c1 = new Coord(2, 2);
            var c2 = new Coord(2, 2);
            var c3 = new Coord();
            var c4 = new Coord();
            Assert.Equal(c1, c2);
            Assert.Equal(c3, c4);
            Assert.NotEqual(c1, c3);
        }
    }
}

File: projects/java/map2d/src/test/java/net/ponderingprogrammer/map2d/CoordTest.java
package net.ponderingprogrammer.map2d;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

public class CoordTest {
    @Test
    @DisplayName("Coord equality test")
    public void testEquality() {
        var c1 = new Coord(2, 2);
        var c2 = new Coord(2, 2);
        var c3 = new Coord();
        var c4 = new Coord();
        Assertions.assertEquals(c1, c2);
        Assertions.assertEquals(c3, c4);
        Assertions.assertNotEquals(c1, c3);
    }
}
